test: add cleanup for web worker instances in test files

ArnavK-09 opened the pull request:
Ensure proper cleanup of web worker instances by adding `await worker.kill()` at the end of each test. This prevents potential memory leaks and ensures tests run in a clean state.

ArnavK-09 committed:
test: add cleanup for web worker instances in test files

Ensure proper cleanup of web worker instances by adding `await worker.kill()` at the end of each test. This prevents potential memory leaks and ensures tests run in a clean state.

ArnavK-09 committed:
chore

seveibar reviewed (commented):
this won't work when tests fail, i think there might be another way to do it, some ideas:

1. Whenever we create a webworker, check a global singleton and kill any others that exist
2. Alt: Use bunfig.toml with a preload directive (e.g. https://github.com/tscircuit/autorouting/blob/main/bunfig.toml) and use ``afterEach` to clean up any web workers that were created (also requires global tracking)

wdyt? If we do (1) I believe we will fix some of the OOM issues I sometimes get when using the CLI

seveibar reviewed (approved):
you can merge if this is blocking, but we should revert and do global webworker tracking imo

ArnavK-09 commented:
> this won't work when tests fail, i think there might be another way to do it, some ideas:
> 
> 1. Whenever we create a webworker, check a global singleton and kill any others that exist
> 2. Alt: Use bunfig.toml with a preload directive (e.g. https://github.com/tscircuit/autorouting/blob/main/bunfig.toml) and use ``afterEach` to clean up any web workers that were created (also requires global tracking)
> 
> wdyt? If we do (1) I believe we will fix some of the OOM issues I sometimes get when using the CLI

Yup your right, lemme look into it

ArnavK-09 committed:
refactor(tests): update import paths and remove redundant worker kill calls

Update import paths for `createCircuitWebWorker` from "lib" to "." to reflect the correct module location. Remove redundant `await circuitWebWorker.kill()` calls in test files as they are no longer necessary.

ArnavK-09 committed:
Revert "refactor(tests): update import paths and remove redundant worker kill calls"

This reverts commit 1a8e30af5083b741b12b85782ae4409f71c4cbb8.

ArnavK-09 committed:
Merge branch 'main' into X2
